### ISSUE TITLE
Add syntax highlighting

### ISSUE THREAD

**Issue by ihollander** [learn-co-curriculum/github-to-canvas#4]
The HTML produced for preformatted code blocks doesn't allow for syntax highlighting. To add this enhancement, we have a couple options:

**Option 1**: Simply use the language from the fenced code blocks to add a class to the \<pre\> tags, and then use a client-side library like [Prism](https://prismjs.com/) to parse the code blocks and add classes required for styling.

This would require a very small change to the gem:

```rb
# lib/github-to-canvas/repository_converter.rb
class CustomRender < Redcarpet::Render::HTML
  def block_code(code, lang)
    # add a class 
    "<pre class='language-#{lang}'>" \
      "<code>#{multi_line(code)}</code>" \
    "</pre>"
  end
```

This would also require adding additional CSS in Canvas to handle the syntax highlighting, which we can use Prism to generate/customize.

**Option 2**: Parse the code blocks at the time the HTML is generated using a tool like [Rouge](https://github.com/rouge-ruby/rouge), so no client-side processing is necessary. This would require adding an additional dependency.

Aside from the dependency, the change to the actual code is minimal:

```rb
# lib/github-to-canvas/repository_converter.rb
require 'rouge'
require 'rouge/plugins/redcarpet'

class CustomRender < Redcarpet::Render::HTML
  include Rouge::Plugins::Redcarpet
  # no need to handle multi-line code with the custom renderer any more!
end
```

This would also require adding additional CSS in Canvas to handle the syntax highlighting. Rouge generates CSS classes that can be styled with a stylesheet generated using [Pygments](https://pygments.org/).

To decide the best option:

- Use both approaches to generate example HTML in all languages covered in Flatiron curriculum: HTML, CSS, JavaScript, JSX, SQL, Ruby, Python, Bash, ...
- Generate some example CSS
- Experiment in Canvas

**Comment by ihollander** [comment]
Demo for Rouge:

```rb
require 'redcarpet'
require 'rouge'
require 'rouge/plugins/redcarpet'
require 'pry'

class CustomRender < Redcarpet::Render::HTML
  include Rouge::Plugins::Redcarpet
end

markdown = <<~MD
# Hello

```js
function hello(world) {
  console.log(world);
}
```

```jsx
function App({ props }) {
  return (
    <div>
      {/* comment */}
      <Header name={props.name} />
    </div>
  )
}
```

```html
<h1>Hello</h1>
<h2>World</h2>
```

```rb
redcarpet = Redcarpet::Markdown.new(CustomRender, tables: true, autolink: true, fenced_code_blocks: true, disable_indented_code_blocks: true)
html = redcarpet.render(markdown)
```

```css
body {
  color: red;
}
```
MD

redcarpet = Redcarpet::Markdown.new(CustomRender, tables: true, autolink: true, fenced_code_blocks: true, disable_indented_code_blocks: true)
puts redcarpet.render(markdown)
```

**Comment by ihollander** [comment]
Created a branch to test with:

https://github.com/learn-co-curriculum/github-to-canvas/tree/syntax-highlighting

To install the gem from the branch, first uninstall current version of the gem locally (all versions):

```sh
gem uninstall github-to-canvas
```

Then:

```sh
git clone git@github.com:learn-co-curriculum/github-to-canvas.git
git checkout syntax-highlighting
gem build github-to-canvas.gemspec
gem install github-to-canvas-0.1.16.gem
```

This will build and install the gem on your system using the code from the branch. Then any time you use `github-to-canvas` locally you can see what effect the new version has on the output.

**Comment by ihollander** [comment]
Closed by #7
